guides/web-apps/sessions/README.md

```diff
@@ -1,12 +1,20 @@
 # Sessions
 
-Sessions are used to retain the client state in your application. A session is represented by an instance of a `Session` class. Session is created on current Starcounter scheduler and should be operated only on that scheduler. That's why one should never store session objects statically (same as one shouldn't store SQL enumerators statically) or use session objects in multithreaded/asynchronous programming. In order to save session and utilize it later please use `Session.SessionId` described below.
+Sessions are used to retain the state in your app. A session is represented by an instance of a `Session` class. The `Json` tree that represents your apps' view-model must be assigned to a session to allow the client-server synchronization of the view.
 
-Sessions and JSON objects can be connected: on JSON object session can be accessed using `Session` property, and on `Session` object, JSON object can be accessed through `Data` property.
+## Session Instance
 
-New session is created by calling constructor on `Session` class. Current session (can be accessed by calling static `Session.Current` property) is automatically set during lifetime of the user handler execution. Whenever session object is set on JSON object using `Session` property (or JSON object is set on session using `Data` property), the `Session.Current` is automatically being set to that object.
+New session is created by calling the constructor on `Session` class. At this point, the static property `Session.Current` is automatically set to your new session and is available during the lifetime of the user handler execution.
 
-Current session is determined and set automatically before user handler is called. Presence of session is determined by one of the ways described below.
+When multiple apps are running, only one of them needs to create the `Session` object, because the session identifies the browser tab, not only your app in that tab. Therefore, you always need to check before creating a new session if `Session.Current` is `null`.
+
+> **Note**: Each browser tab is a separate state of the UI. Therefore, each tab is tied to its own session. This makes it totally different from the session concept in frameworks like ASP.NET or Zend, where a session means all browser tabs created by the same user.
```

Comment:
where a session means all browser tabs -> where a session stores data from all the browser tabs